src/main/java/com/google/devtools/build/lib/rules/proto/ProtoConfiguration.java

```diff
@@ -243,6 +246,20 @@ public boolean runExperimentalProtoExtraActions() {
     return options.experimentalProtoExtraActions;
   }
 
+  @SkylarkConfigurationField(
+      // Must match the value of `_protoc_key`
+      // in `@rules_proto//proto/private/rules:proto_toolchain.bzl`.
+      name = "protoc_do_not_use_or_we_will_break_you_without_mercy",
```

Comment:
What's the plan for this command line option?

The usual thing we do is to have rules like `CcToolchainAliasRule`, although those can just as well be implemented in Starlark. That way, you don't need to have a late-bound default. Even better, you can prescribe a particular name for that rule so that people use it as little as possible (although we don't do that for the rest of the alias rules)

Reply:
All of these options here are meant for use in a new `proto_toolchain` rule that is eventually going to replace the command-line options.

i.e. Now, `proto_toolchain` exposes the options and looks like this: https://github.com/bazelbuild/rules_proto/blob/919671ef0f73f5a82037eaabedc39c938c13cf60/proto/BUILD#L8, and eventually, it will look like this, and the command-line options are deprecated and removed:

```python
proto_toolchain(
    name = "default_toolchain_impl",
    compiler = "@com_google_protobuf//:protoc",
    protocopt = ["--foo", "--bar"],
    strict_deps = "ERROR",
    visibility = ["//visibility:public"],
)
```

I guess we could also implement `proto_toolchain` as native rule first, but since we're going to replace it with a Starlark rule eventually, we can also implement it Starlark now.

Reply:
Oh my question I already sent is answered here :) So you plan to replace command line options completely?